src/Symfony/Component/Validator/Mapping/Loader/XmlFileLoader.php

```diff
@@ -179,6 +184,8 @@ protected function parseFile(string $path): \SimpleXMLElement
 
     private function loadClassesFromXml()
     {
+        parent::__construct($this->file);
```

Comment:
I would prefer not to call the constructor here but move the logic that is located there in the base class into a dedicated method that we can call here then.

Reply:
that'd mean opening an API for that :(

Reply:
We can mark the method as internal.

Reply:
we can, but what's the issue with calling the constructor?

Reply:
IIRC, calling our constructors as non-constructor methods is not covered by our BC policy.

Here, you are still opening an API without telling it.

Reply:
I'm not opening anything, the constructor of the base class won't change in a non-BC way

Reply:
In case that's what is bothering you, the parent constructor is still going to be called only once.

Reply:
That’s okay, it just doesn’t feel right to me to call the constructor at a later time.

Reply:
Not sure why though. ProxyManager does it all the time for example. Lazy initialization is legit concern.